Add io.github.buonhobo.KatharaGUI

BuonHobo opened the pull request:
<!-- ⚠️  The submission PR must be against the `new-pr` branch ⚠️  -->

### Please confirm your submission meets all the criteria

<!-- Please replace each `[ ]` with `[X]` when the step is complete -->

- [X] Please describe the application briefly.
  <!-- insert the description here -->
KatharaGUI is a simple GUI for the Kathara network emulation tool. 
It allows you to start and stop labs and connect to the containers in a convenient way. 
KatharaGUI requires Docker to be installed and correctly configured on your system. 
This app is written in Python and uses GTK 4, libadwaita, and VTE.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**


<!-- ⚠️  Don't modify anything below this line ⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

bbhtt commented on io.github.buonhobo.KatharaGUI.json:
```diff
@@ -0,0 +1,64 @@
+          "commands": ["bash -c 'python3 /app/bin/KatharaGUI.py'"],
+          "dest-filename": "KatharaGUI.sh"
+        },
+        {
+          "type": "inline",
+          "dest-filename": "io.github.buonhobo.KatharaGUI.desktop",
+          "contents": "[Desktop Entry]\nVersion=1.0\nType=Application\nName=KatharaGUI\nExec=KatharaGUI.sh\nComment=A simple Kathara GUI\nIcon=io.github.buonhobo.KatharaGUI\nTerminal=false\nCategories=Utility;Application;"
+        }
```
Have a desktop file in your github repo and use that, it's a universal standard.

bbhtt commented on io.github.buonhobo.KatharaGUI.json:
```diff
@@ -0,0 +1,64 @@
+        },
+        {
+          "type": "script",
+          "commands": ["bash -c 'python3 /app/bin/KatharaGUI.py'"],
```
No need for bash -c

bbhtt replied:
Add the python shebang to the file, then there is no need for this script as well.

bbhtt commented on io.github.buonhobo.KatharaGUI.metainfo.xml:
metainfo file should be upstream not here.

bbhtt commented on io.github.buonhobo.KatharaGUI.metainfo.xml:
```diff
@@ -0,0 +1,72 @@
+  <screenshots>
+    <screenshot type="default">
+      <caption>KatharaGUI when connected to a machine</caption>
+      <image>https://github.com/BuonHobo/KatharaGUI/blob/main/data/screenshots/LFGFFaB.png?raw=true</image>
```
these needs to be direct raw.githubusercontet.com links no html pages. Use a commit or a tag in place of `main` in the link.

bbhtt replied:
same for below.

BuonHobo commented:
Thank you  very much @bbhtt for your quick and helpful support.
I addressed all of the comments:
- I put a desktop file in my upstream repo
- Added the python shebang to eliminate the need for a wrapper script
- Upstreamed the metainfo file
- Changed the screenshot links back to direct imgur links (which they were in the first place, I had changed them to github links thinking it would be better)

Let me know if it's all fine now

bbhtt commented:
Looks like you rebased over master branch or merged master branch of this repo. Can you fix it ?

bbhtt commented on io.github.buonhobo.KatharaGUI.json:
```diff
@@ -0,0 +1,53 @@
+    {
+      "name": "vte",
+      "buildsystem": "meson",
+      "config-opts": ["--buildtype=release", "-Dgtk3=false", "-Dgtk4=true"],
```
```suggestion
      "config-opts": [
        "--buildtype=release",
        "-Dgtk3=false",
        "-Dgtk4=true"
      ],
```

BuonHobo commented:
Oh you are right. Is it alright if I just delete the .github directory?

bbhtt commented:
the commits are a bit messy still, try checking out your branch and do `git rebase --onto new-pr`

BuonHobo commented:
I think it's fixed now. Sorry for wasting your time

bbhtt commented:
You can start the build

bbhtt commented:
bot, build io.github.buonhobo.KatharaGUI

BuonHobo commented:
This build seems to work fine on my computer. 
If you want to test it out you have to install docker and add yourself to the docker group. 
In order to start a lab, you need to select a directory with a lab.conf file inside it, this is a suitable option:
https://github.com/KatharaFramework/Kathara-Labs/tree/main/main-labs/labs-integrating-several-technologies/small-internet-with-dns-and-web-server/kathara-lab_small-internet-with-dns-and-web-server

In that repo there are many more labs you can play with. Let me know if you have any issues

bbhtt commented on io.github.buonhobo.KatharaGUI.json:
```diff
@@ -0,0 +1,57 @@
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--filesystem=/run/docker.sock",
+    "--filesystem=home",
```
why do you need home ? GTK4 filechoosers should use the portal by default.

bbhtt commented on io.github.buonhobo.KatharaGUI.json:
```diff
@@ -0,0 +1,57 @@
+  "sdk": "org.gnome.Sdk",
+  "command": "KatharaGUI",
+  "finish-args": [
+    "--device=all",
+    "--device=dri",
```
which one is needed ?

BuonHobo commented:
I changed the permissions, now I only have:

  "finish-args": [
    "--socket=fallback-x11",
    "--socket=wayland",
    "--device=dri",
    "--filesystem=/run/docker.sock"
  ],

bbhtt commented:
You have to keep `share=ipc` that one is needed when running on X11 sessions.

BuonHobo commented:
I added it back

bbhtt commented:
bot, build io.github.buonhobo.KatharaGUI

BuonHobo commented:
Would it be possible to rename the app to Katharsis before it's pushed to flathub?

bbhtt commented:
The ID? It has to match your GitHub repo name, so you have to rename that too.

Changing the `nane` tag is fine.
